### ISSUE TITLE
flatpak: Marker (markdown editor) does not start up

### ISSUE THREAD

**Issue by alexeymuranov** [NixOS/nixpkgs#53441]
## Issue description

I cannot run [Marker](https://fabiocolacio.github.io/Marker/) markdown editor installed with `flatpak` [from Flathub](https://beta.flathub.org/apps/details/com.github.fabiocolacio.marker):
```console
$ flatpak run com.github.fabiocolacio.marker

(marker:2): Gtk-WARNING **: 13:04:24.099: Locale not supported by C library.
	Using the fallback 'C' locale.

(marker:2): GdkPixbuf-WARNING **: 13:04:24.125: Cannot open pixbuf loader module file '/nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory

This likely means that your installation is broken.
Try running the command
  gdk-pixbuf-query-loaders > /nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
to make things work again for the time being.

(marker:2): GdkPixbuf-WARNING **: 13:04:24.197: Cannot open pixbuf loader module file '/nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory

This likely means that your installation is broken.
Try running the command
  gdk-pixbuf-query-loaders > /nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
to make things work again for the time being.

(process:11): Gtk-WARNING **: 13:04:24.259: Locale not supported by C library.
	Using the fallback 'C' locale.

(WebKitWebProcess:11): GdkPixbuf-WARNING **: 13:04:24.284: Cannot open pixbuf loader module file '/nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory

This likely means that your installation is broken.
Try running the command
  gdk-pixbuf-query-loaders > /nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
to make things work again for the time being.

(marker:2): GdkPixbuf-WARNING **: 13:04:24.590: Cannot open pixbuf loader module file '/nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory

This likely means that your installation is broken.
Try running the command
  gdk-pixbuf-query-loaders > /nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
to make things work again for the time being.

(marker:2): Gtk-WARNING **: 13:04:24.590: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.

(marker:2): GdkPixbuf-WARNING **: 13:04:24.593: Cannot open pixbuf loader module file '/nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory

This likely means that your installation is broken.
Try running the command
  gdk-pixbuf-query-loaders > /nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
to make things work again for the time being.

(marker:2): GdkPixbuf-WARNING **: 13:04:24.593: Cannot open pixbuf loader module file '/nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory

This likely means that your installation is broken.
Try running the command
  gdk-pixbuf-query-loaders > /nix/store/mfwiwwqvmsxqpmhvq61aqivvkaqzc1bm-librsvg-2.42.4/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
to make things work again for the time being.
**
Gtk:ERROR:gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Adwaita/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
```

### Steps to reproduce

On NixOS 18.09, enable `flatpak` in `/etc/nixos/configuration.nix`:
```nix
{ # ...
  services.flatpak.enable = true;
}
```
Add the Flathub repository:
```console
$ flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
```
Install Marker from Flathub and run it:
```console
$ flatpak install flathub com.github.fabiocolacio.marker
$ flatpak run com.github.fabiocolacio.marker
```

## Technical details

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.13, NixOS, 18.09.1834.9d608a6f592 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(alexey): `"nixos-17.03-17.03.1949.78e9665b48f, nixos-17.09-17.09.3238.e984f9e48e1"`
 - channels(root): `"nixos-18.09.1834.9d608a6f592"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by jtojnar** [comment]
What is the version of flatpak and what channel did you install it from? It is quite weird, looks like `GDK_PIXBUF_FILE` environment variable is leaking to the container.

**Comment by alexeymuranov** [comment]
I have
```console
$ flatpak --version
Flatpak 1.0.5
```

It is installed by setting `services.flatpak.enable = true;` in `/etc/nixos/configuration.nix`, so i suppose it is from `release-18.09` channel.

**Comment by hedning** [comment]
Indeed, http://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-run :
> Environment variables are generally passed on to the sandboxed application, with certain exceptions.

I guess we have two options here (how to apply them once):
1. mount the nix store (`flatpak run --filesystem=/nix/store:ro com.github.fabiocolacio.marker`)
2. unset `GDK_PIXBUF_MODULE_FILE` env variable (flatpak run --env=GDK_PIXBUF_MODULE_FILE= com.github.fabiocolacio.marker`)

Unsetting the environment variable seems like the least offensive thing to do here, and could probably be accepted by upstream as the gnome sdk includes the cache.

**Comment by alexeymuranov** [comment]
Indeed, this is fixed now in 19.03, thanks.
